patroni/postgresql/sync.py

```diff
@@ -1,23 +1,163 @@
 import logging
 import re
+import time
+
+from copy import deepcopy
 
 from .validator import CaseInsensitiveDict
 from ..psycopg import quote_ident as _quote_ident
 
 logger = logging.getLogger(__name__)
 
 SYNC_STANDBY_NAME_RE = re.compile(r'^[A-Za-z_][A-Za-z_0-9\$]*$')
+SYNC_REP_PARSER_RE = re.compile(r"""
+           (?P<first> [fF][iI][rR][sS][tT] )
+         | (?P<any> [aA][nN][yY] )
+         | (?P<space> \s+ )
+         | (?P<ident> [A-Za-z_][A-Za-z_0-9\$]* )
+         | (?P<dquot> " (?: [^"]+ | "" )* " )
+         | (?P<star> [*] )
+         | (?P<num> \d+ )
+         | (?P<comma> , )
+         | (?P<parenstart> \( )
+         | (?P<parenend> \) )
+         | (?P<JUNK> . )
+        """, re.X)
+_EMPTY_SSN = {'type': 'off', 'num': 0, 'members': [], 'cimembers': {}}
 
 
 def quote_ident(value):
     """Very simplified version of quote_ident"""
     return value if SYNC_STANDBY_NAME_RE.match(value) else _quote_ident(value)
 
 
+def parse_sync_standby_names(value):
+    """Parse postgresql synchronous_standby_names to constituent parts.
+    Returns dict with the following keys:
```

Comment:
I am also wondering, if `members` will be used in the future or only `cimembers` has actual value outside this function?
am asking because the following looks a bit confusing:
```
>>> parse_sync_standby_names('ANY 3(A, a, B)')
 {'type': 'quorum', 'num': 3, 'members': ['A', 'a', 'B'], 'cimembers': {'a': True, 'B': True}}
 ```
 btw, is `num==3` actually correct in this case? (not sure, what it refers to)

Reply:
You are right, we don't need two values.
All comparisons between member names and application_names from pg_stat_replication should be case insensitive, therefore we will just use CaseInsensitiveDict() and when (in the future) the content will be required we can just return list(.keys())